### PR TITLE
added feature/CC-101.2-add-new-customer-dispatch-call for adding new customer

### PR DESCRIPTION
**Pull request by ankit-mogha**
What does this change do?
add a dispatch call to add a new customer.

Any additional information for the reviewer to start
NA

How should this be manually tested?
We need the window OS in which Rust is installed. then execute this rust program

Which Trello task and sub-task does this change relate to?
CC - SP2 101.2 - Add dispatch call for add_customer.

Are there any changes pending?
No

Does any team have to be notified of changes in this feature?
Yes

Definition of Done:
- [ ] Is there >90% unit test code coverage?
- [ ] Does this PR add new dependencies? If so, please list out the same.
- [ ]  Will this feature require a new piece of infrastructure to be implemented?
- [x] Is there appropriate logging included?
- [x] Does the project compile ok?
- [x] Have Clippy violations been fixed?
- [x] Have Code is properly formatted?

### CODE REVIEW

File: pallets/carpooling/src/lib.rs [hunk 1]
```diff
@@ -17,90 +16,98 @@ mod benchmarking;
...
     #[pallet::getter(fn get_driver)]
-    pub type Driver<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, DriverOf<T>>;
-	#[pallet::storage]
+    pub type Driver<T: Config> = StorageMap<_, Blake2_128Concat, u32, DriverOf<T>>;
+    #[pallet::storage]
     #[pallet::getter(fn get_booking)]
-    pub type Booking<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, T::AccountId>;
-	// Learn more about declaring storage items:
-	// https://substrate.dev/docs/en/knowledgebase/runtime/storage#declaring-storage-items
+    pub type Booking<T: Config> = StorageMap<_, Blake2_128Concat, u32, T::AccountId>;
```
Comment (aman2909verma) [inline pallets/carpooling/src/lib.rs:61]: change T::AccountId to u32

File: pallets/carpooling/src/lib.rs [hunk 2]
```diff
@@ -17,90 +16,98 @@ mod benchmarking;
...
+    use frame_support::{dispatch::DispatchResult, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    // CustomerOf is datatype used for storage in Customer
+    type CustomerOf<T> = SCustomer<<T as frame_system::Config>::Hash>;
+    // SCustomer is a struct for Customer
     #[derive(Encode, Decode, Clone, Default, PartialEq, RuntimeDebug)]
     pub struct SCustomer<Hash> {
         pub id: u32,
         pub name: Hash,
         pub location: Hash,
```
Comment (aman2909verma) [inline pallets/carpooling/src/lib.rs:28]: make the location as tuple of (u32,u32)

File: pallets/carpooling/src/lib.rs [hunk 3]
```diff
@@ -17,90 +16,98 @@ mod benchmarking;
...
-	type DriverOf<T> = SDriver<<T as frame_system::Config>::Hash>;
+    // DriverOf is datatype used for storage in Driver
+    type DriverOf<T> = SDriver<<T as frame_system::Config>::Hash>;
     //SDriver is a struct for Driver
-	#[derive(Encode, Decode, Clone, Default, PartialEq, RuntimeDebug)]
+    #[derive(Encode, Decode, Clone, Default, PartialEq, RuntimeDebug)]
     pub struct SDriver<Hash> {
         pub id: u32,
         pub car_no: Hash,
         pub location: Hash,
```
Comment (aman2909verma) [inline pallets/carpooling/src/lib.rs:37]: make the location as tuple of (u32,u32)

File: pallets/carpooling/src/tests.rs [hunk 4]
```diff
@@ -1,23 +1,70 @@
...
+use frame_support::assert_ok;
+use frame_support::sp_runtime::DispatchError::Module;
+use sp_core::Encode;
+use sp_runtime::traits::Hash;
 
 #[test]
-fn it_works_for_default_value() {
-	new_test_ext().execute_with(|| {
-		// Dispatch a signed extrinsic.
-		assert_ok!(TemplateModule::do_something(Origin::signed(1), 42));
```
Comment (aman2909verma) [inline pallets/carpooling/src/tests.rs:8]: use Carpooling not TemplateModule in all test cases

File: pallets/carpooling/src/tests.rs [hunk 5]
```diff
@@ -1,23 +1,70 @@
...
-		// Read pallet storage and assert an expected result.
-		assert_eq!(TemplateModule::something(), Some(42));
-	});
+fn add_new_customer() {
+    new_test_ext().execute_with(|| {
+        let new_cust = SCustomer {
+            id: 32,
+            name: "Ankit".using_encoded(<Test as frame_system::Config>::Hashing::hash),
+            location: "New Delhi".using_encoded(<Test as frame_system::Config>::Hashing::hash),
```
Comment (aman2909verma) [inline pallets/carpooling/src/tests.rs:13]: change location to a tuple in all test cases for exact location

**Review by aman2909verma — CHANGES_REQUESTED** [review]

File: pallets/carpooling/src/tests.rs [hunk 6]
```diff
@@ -1,23 +1,70 @@
...
+        };
+        // Dispatch a signed extrinsic.
+        assert_eq!(
+            TemplateModule::add_new_customer(Origin::signed(1), 42, new_cust),
```
Comment (aman2909verma) [inline pallets/carpooling/src/tests.rs:17]: use Carpooling not TemplateModule in all test cases

**Review by aman2909verma — CHANGES_REQUESTED** [review]

**Review by aman2909verma — APPROVED** [review]

**Review by Ayushi-knoldus — APPROVED** [review]

File: pallets/carpooling/src/lib.rs [hunk 7]
```diff
@@ -17,90 +16,96 @@ mod benchmarking;
...
+
+    // Dispatchable functions allows users to interact with the pallet and invoke state changes.
+    // These functions materialize as "extrinsics", which are often compared to transactions.
+    // Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// An example dispatchable that takes a singles value as a parameter, writes the value to
+        /// storage and emits an event. This function must be dispatched by a signed extrinsic.
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn add_new_customer(
```
Comment (bishtpawan) [inline pallets/carpooling/src/lib.rs:91]: please add documentation

File: pallets/carpooling/src/lib.rs [hunk 8]
```diff
@@ -17,90 +16,96 @@ mod benchmarking;
...
-	impl<T:Config> Pallet<T> {
-		/// An example dispatchable that takes a singles value as a parameter, writes the value to
-		/// storage and emits an event. This function must be dispatched by a signed extrinsic.
-		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
-		pub fn do_something(origin: OriginFor<T>, something: u32) -> DispatchResult {
-			// Check that the extrinsic was signed and get the signer.
-			// This function will return an error if the extrinsic is not signed.
-			// https://substrate.dev/docs/en/knowledgebase/runtime/origin
-			let who = ensure_signed(origin)?;
+            match <Customer<T>>::get(cust_id) {
```
Comment (bishtpawan) [inline pallets/carpooling/src/lib.rs:98]: please try to update flow

**Pull request closed without merging**